pkg/test/pods.go

```diff
@@ -143,6 +145,25 @@ func Pod(overrides ...PodOptions) *v1.Pod {
 			Phase:      options.Phase,
 		},
 	}
+	// If PreStopSleep is enabled, add it to each of the containers.
+	// Can't use v1.LifecycleHandler == v1.SleepAction as that's a feature gate in Alpha 1.29.
+	// https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#hook-handler-implementations
+	if options.PreStopSleep != nil {
```

Comment:
I'm assuming that this is put here for the E2E test suite? I didn't see a functional reason why this would need to be part of the functional testing suite

Reply:
Yeah, that's right